### DISCUSSION TITLE
Add Support For Home Assistant Users

### DISCUSSION THREAD

**Discussion by KazWolfe** [hassio-addons/addon-grafana#363] [Ideas]
# Problem/Motivation

I would like to expose certain Grafana dashboards to certain Home Assistant users, and otherwise enforce certain permissions so that other users of my Home Assistant instance cannot alter or delete dashboards I have made. Ideally, I would be able to do this without disabling Home Assistant's ingress or built-in authentication.

## Proposed changes

I manged to get this working (possibly?) by setting the `grafana_ingress_user` variable to `$http_x_remote_user_name`. This leverages the fact that Grafana is [already configured](https://github.com/hassio-addons/addon-grafana/blob/main/grafana/rootfs/etc/grafana/grafana.ini#L203) to use the [auth proxy](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/) feature, and simply passing it the [user's credentials from Ingress](https://developers.home-assistant.io/docs/add-ons/security#authenticating-a-user-when-using-ingress).

It seems to work at least tolerably, with Grafana having created a new admin user for me automatically upon first login:

![image](https://github.com/hassio-addons/addon-grafana/assets/5192145/3ce3dfd0-f5e4-467c-962c-08962c38142c)

I'm unsure how well this capability would scale, nor if there are any other side effects to this, but it does look like this could pave the way to HASS users getting special permissions in Grafana as well.

Is this a viable strategy moving forward for this use case, and is there anything that the configuration can do to better handle this case (e.g. reading from `X-Remote-User-Name` directly?)

**Comment by frenck** [comment]
Converted to a discussion, as it is not an issue but a feature request.

I have no plans on adding support for that. Feel free to open a PR that does so (without breaking existing installations).

**Reply by KazWolfe** [reply]
Ah, apologies. I'll take a look and hack around to see how this will behave on a fresh install, as that would seem like it can be the biggest hiccup here.
